src/main/java/com/google/api/codegen/discovery/transformer/java/JavaSurfaceTransformer.java

```diff
@@ -0,0 +1,53 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery.transformer.java;
+
+import com.google.api.codegen.ApiaryConfig;
+import com.google.api.codegen.discovery.transformer.MethodToViewTransformer;
+import com.google.api.codegen.discovery.transformer.MethodTypeTable;
+import com.google.api.codegen.discovery.transformer.SurfaceTransformerContext;
+import com.google.api.codegen.discovery.viewmodel.StaticLangXApiView;
+import com.google.api.codegen.util.java.JavaTypeTable;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.protobuf.Method;
+
+/*
+ * Transforms a Model into the standard discovery surface in Java.
+ */
+public class JavaSurfaceTransformer implements MethodToViewTransformer {
```

Comment:
I recommend calling this JavaDiscoverySampleTransformer. The term "surface" refers to an API surface; this is generating code to call a surface (from Apiary).